editor.tsx

```diff
@@ -3,29 +3,102 @@ import { ASSET_CACHE_BUST_KEY } from "$fresh/runtime.ts";
 import { renderToString } from "preact-render-to-string";
 import LiveContext from "./context.ts";
 import getSupabaseClient, { getSupabaseClientForUser } from "./supabase.ts";
-import { Flag, Module } from "./types.ts";
+import type {
+  Flag,
+  Module,
+  PageComponentData,
+  PageDataData,
+  PageLoaderData,
+} from "./types.ts";
 import {
   componentNameFromPath,
   getComponentModule,
 } from "./utils/component.ts";
 import { createServerTiming } from "./utils/serverTimings.ts";
 import { duplicateProdPage, getFlagFromPageId } from "./utils/supabase.ts";
+import type { JSONSchema7 } from "json-schema";
+import {
+  generateLoaderInstance,
+  loaderInstanceToProp,
+  loaderPathToKey,
+} from "./utils/loaders.ts";
+import type { LivePageData } from "./live.tsx";
 
 const ONE_YEAR_CACHE = "public, max-age=31536000, immutable";
 
-const updateDraft = async (req: Request, url: URL, ctx: any) => {
-  const { components, template, siteId, variantId, experiment } = ctx;
+/**
+ * @description This function creates a loaders list based on component props that depends on Loaders.
+ *  Also, mutate prop that depends on Loaders to component[index].props.[propName] = `loadername-uuid`
+ */
+const generateLoadersFromComponents = (
+  _: Request,
+  __: URL,
+  ctx: any,
+): PageDataData => {
+  const { components: ctxComponents } = ctx;
+  const components = JSON.parse(JSON.stringify(ctxComponents));
```

Comment:
I've benchmarked structuredClone vs JSON.parse+JSON.stringfy. The second option is faster. Take a look at the results:
```
❯ deno bench --unstable bench_clone.ts
Check file:///home/igor/dev/deco/hortifruti/bench_clone.ts
cpu: AMD Ryzen 7 1800X Eight-Core Processor
runtime: deno 1.26.1 (x86_64-unknown-linux-gnu)

file:///home/igor/dev/deco/hortifruti/bench_clone.ts
benchmark            time (avg)             (min … max)       p75       p99      p995
------------------------------------------------------- -----------------------------
StructuredClone   25.32 µs/iter  (22.66 µs … 187.52 µs)  23.78 µs  64.68 µs 101.37 µs
JSON Stringify    14.32 µs/iter  (13.46 µs … 175.92 µs)  13.84 µs  29.59 µs  31.92 µs
```

the benchmark test:
```ts
const data = [
  { component: "ProductShelf", props: { title: "Lucis", products: { collection: "arroz" }, too: { deep: { nested: 1 } }}, },
{ component: "ProductShelf", props: { title: "Lucis", products: { collection: "arroz" }, too: { deep: { nested: 1 } }}, },
{ component: "ProductShelf", props: { title: "Lucis", products: { collection: "arroz" }, too: { deep: { nested: 1 } }}, },
{ component: "ProductShelf", props: { title: "Lucis", products: { collection: "arroz" }, too: { deep: { nested: 1 } }}, },
{ component: "ProductShelf", props: { title: "Lucis", products: { collection: "arroz" }, too: { deep: { nested: 1 } }}, },
{ component: "ProductShelf", props: { title: "Lucis", products: { collection: "arroz" }, too: { deep: { nested: 1 } }}, },
 ];
Deno.bench("StructuredClone", () => {
  structuredClone(data);
});
Deno.bench("JSON Stringify", () => {
  JSON.parse(JSON.stringify(data));
});
```